modules/apps/29-fee/types/keys.go

```diff
@@ -71,11 +71,27 @@ func ParseKeyFeeEnabled(key string) (portID, channelID string, err error) {
 	return portID, channelID, nil
 }
 
-// KeyCounterpartyRelayer returns the key for relayer address -> counteryparty address mapping
+// KeyCounterpartyRelayer returns the key for relayer address -> counterparty address mapping
 func KeyCounterpartyRelayer(address, channelID string) []byte {
 	return []byte(fmt.Sprintf("%s/%s/%s", CounterpartyRelayerAddressKeyPrefix, address, channelID))
 }
 
+// ParseKeyCounterpartyRelayer returns the registered relayer address -> counterparty address mapping
+func ParseKeyCounterpartyRelayer(key string, counterpartyAddress string) (RegisteredRelayerAddress, error) {
```

Comment:
I think this function should just parse the key. It should be up to the caller to construct `RegisteredRelayerAddress`

Reply:
Agreed